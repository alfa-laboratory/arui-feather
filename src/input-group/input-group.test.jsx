/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { shallow } from 'enzyme';

import InputGroup from './input-group';
import Input from '../input/input';

describe('input-group', () => {
    it('should render without any children', () => {
        let inputGroup = shallow(<InputGroup />);

        expect(inputGroup).toMatchSnapshot();
    });

    it('should render with only one children', () => {
        let inputGroup = shallow(<InputGroup><Input key='1' /></InputGroup>);

        expect(inputGroup).toMatchSnapshot();
    });

    it('should render with many input children without problems', () => {
        let inputGroup = shallow(<InputGroup><Input key='1' /><Input key='2' /></InputGroup>);

        expect(inputGroup).toMatchSnapshot();
    });
});
