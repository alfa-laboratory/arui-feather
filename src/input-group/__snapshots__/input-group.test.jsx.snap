// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input-group should render with many input children without problems 1`] = `
<span
  className="input-group input-group_theme_alfa-on-white control-group"
  role="group"
  tabIndex="-1"
>
  <span
    className="input-group__input-case"
    key="input-0/.0"
  >
    <Input
      formNoValidate={false}
      key="1"
      size="m"
      type="text"
    />
  </span>
  <span
    className="input-group__input-case"
    key="input-1/.0"
  >
    <Input
      formNoValidate={false}
      key="2"
      size="m"
      type="text"
    />
  </span>
</span>
`;

exports[`input-group should render with only one children 1`] = `
<span
  className="input-group input-group_theme_alfa-on-white control-group"
  role="group"
  tabIndex="-1"
>
  <span
    className="input-group__input-case"
    key="input-0/.0"
  >
    <Input
      formNoValidate={false}
      key="1"
      size="m"
      type="text"
    />
  </span>
</span>
`;

exports[`input-group should render without any children 1`] = `
<span
  className="input-group input-group_theme_alfa-on-white control-group"
  role="group"
  tabIndex="-1"
/>
`;
